examples/prom/linear_elasticity_global_rom.cpp

```diff
@@ -0,0 +1,525 @@
+    const char* device_config = "cpu";
+    bool visit = true;
+    bool fom = false;
+    bool offline = false;
+    bool merge = false;
+    bool online = false;
+    int precision = 8;
+    int id = 0;
+    int nsets = 0;
+    double coef = 1.0;
+    double ext_force = -1.0e-2;
+    double E = 2.5;
+    double nu = 0.25;
+
+
+    OptionsParser args(argc, argv);
+    args.AddOption(&mesh_file, "-m", "--mesh",
+        "Mesh file to use.");
+    args.AddOption(&order, "-o", "--order",
+        "Finite element order (polynomial degree).");
+    args.AddOption(&id, "-id", "--id", "Parametric id");
+    args.AddOption(&nsets, "-ns", "--nset", "Number of parametric snapshot sets");
+    args.AddOption(&static_cond, "-sc", "--static-condensation", "-no-sc",
+        "--no-static-condensation", "Enable static condensation.");
+    args.AddOption(&ext_force, "-f", "--ext-force",
+        "External force applied at end.");
+    args.AddOption(&E, "-E", "--youngs-modulus",
+        "Youngs modulus.");
+    args.AddOption(&nu, "-nu", "--poissons-ratio",
+        "Poisson's ratio.");
+    args.AddOption(&amg_elast, "-elast", "--amg-for-elasticity", "-sys",
+        "--amg-for-systems",
+        "Use the special AMG elasticity solver (GM/LN approaches), "
+        "or standard AMG for systems (unknown approach).");
+    args.AddOption(&reorder_space, "-nodes", "--by-nodes", "-vdim", "--by-vdim",
+        "Use byNODES ordering of vector space instead of byVDIM");
+    args.AddOption(&device_config, "-d", "--device",
+        "Device configuration string, see Device::Configure().");
+    args.AddOption(&visit, "-visit", "--visit-datafiles", "-no-visit",
+        "--no-visit-datafiles",
+        "Save data files for VisIt (visit.llnl.gov) visualization.");
+    args.AddOption(&visualization, "-vis", "--visualization", "-no-vis",
+        "--no-visualization",
+        "Enable or disable GLVis visualization.");
+    args.AddOption(&fom, "-fom", "--fom", "-no-fom", "--no-fom",
+        "Enable or disable the fom phase.");
+    args.AddOption(&offline, "-offline", "--offline", "-no-offline", "--no-offline",
+        "Enable or disable the offline phase.");
+    args.AddOption(&online, "-online", "--online", "-no-online", "--no-online",
+        "Enable or disable the online phase.");
+    args.AddOption(&merge, "-merge", "--merge", "-no-merge", "--no-merge",
+        "Enable or disable the merge phase.");
+
+    args.Parse();
+    if (!args.Good())
+    {
+        if (myid == 0)
+        {
+            args.PrintUsage(cout);
+        }
+        return 1;
+    }
+    if (myid == 0)
+    {
+        args.PrintOptions(cout);
+    }
+
+    // 3. Enable hardware devices such as GPUs, and programming models such as
+   //    CUDA, OCCA, RAJA and OpenMP based on command line options.
+    Device device(device_config);
+    if (myid == 0) { device.Print(); }
+
+
+    // 4. Read the (serial) mesh from the given mesh file on all processors.  We
```

Comment:
Indenting is not consistent. In libROM, we run `./scripts/stylize.sh` with astyle 2.05 (or 3.1, we should document this in the README) to ensure consistent formatting. Let us know if you need help with astyle.

Reply:
Thanks! I hadn't seen that before. I'll try it out and let you know.

Reply:
In the latest commit I used `astyle `to format the code.